### PR TITLE
Eslint takeover

### PR DESCRIPTION
**Pull request by skidding**
Integrated ESLint with the Airbnb preset. 😓 

@bogdanstefan @flaviusone @RadValentin wdyt?

### CODE REVIEW

**Comment by flaviusone** [comment]
```js
/* eslint-env browser, mocha */
/* eslint-disable
  global-require,
  no-unused-vars,
  no-unused-expressions,
  import/no-unresolved,
  import/no-extraneous-dependencies
*/
/* global expect, sinon */
```
It seems that you have something like this in almost every test file. Why don't you make a custom eslintrc file just for the test folder that will dedupe a lot of this rules?

**Comment by flaviusone** [comment]
image
image

**Comment by flaviusone** [comment]
LGTM otherwise ¯\_(ツ)_/¯

**Comment by catalinmiron** [comment]
> I know that the airbnb wants it like this but it still hurts 😆.

Hurts because you're not writing javascript ma friend:) 👔 🎉

**Comment by bogdanstefan** [comment]
Just curious, why disable `no-unused-vars` in tests and not remove them?

Anyway, looks beautiful now. 👏

File: test/helpers/render-component.js [hunk 1]
```diff
@@ -1,25 +1,25 @@
-var React = require('react'),
-    ReactDOM = require('../../packages/react-dom-polyfill')(React),
-    $ = require('jquery'),
-    ComponentTree = require('../../packages/react-component-tree'),
-    ComponentPlayground = require('../../packages/react-component-playground/src');
+/* eslint-env browser */
+/* eslint-disable react/no-find-dom-node */
```
Comment (valentin-radulescu-hs) [inline test/helpers/render-component.js]: Can this rule be part of the global config?

File: packages/react-dom-polyfill/test/pre-react-dom.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
+/* eslint-env node, mocha */
+/* eslint-disable react/no-find-dom-node */
+/* global expect, sinon */
```
Comment (valentin-radulescu-hs) [inline packages/react-dom-polyfill/test/pre-react-dom.js]: How about making a tests config and including it for all test folders?

**Comment by valentin-radulescu-hs** [comment]
👍  to what @flaviusone said, maybe you could do something about the annotations? Tests could do with a separate config. Other that that looks good!

**Comment by valentin-radulescu-hs** [comment]
👌

**Pull request merged**